### DISCUSSION TITLE
Error when using plot_parameter_stability

### DISCUSSION THREAD

**Discussion by AinhoaOleaga** [georgebv/pyextremes#26] [Q&A] [answered]
Hello,

When I try to use the function plot_parameter_stability on a pandas.core.series.Series I have a shape problem : 
 -- "x and y must have same first dimension, but have shapes (100,) and (1,)". 
I do not understand why I have this problem as the line of code that I used is :
 -- "pyextremes.plot_parameter_stability(ts = data)"
nothing more nothing less. 
The data is of shape (50241,) but it shouldn't be a problem. 
After the error I have the basis of the plots for shape and modified scale but no plot inside.

Thank you in advance for your help. 

Regards

**Comment by georgebv** [comment, marked as answer]
Please provide exact code and data you used, I cannot say anything until I reproduce your error. And tell me which version of pyextremes you are using.

**Reply by AinhoaOleaga** [reply]
Hello,

I use the version 2.0.0. 
Unfortunately I cannot provide you with the dataset but the error log looks as follows : 

~/.local/lib/python3.7/site-packages/pyextremes/tuning/threshold_selection.py in plot_parameter_stability(ts, thresholds, r, extremes_type, alpha, n_samples, figsize)
    376             color="#F85C50",
    377             lw=2,
--> 378             zorder=15,
    379         )
    380         ax_scale.plot(

/envs/default/lib/python3.7/site-packages/matplotlib/axes/_axes.py in plot(self, scalex, scaley, data, *args, **kwargs)
   1630         """
   1631         kwargs = cbook.normalize_kwargs(kwargs, mlines.Line2D)
-> 1632         lines = [*self._get_lines(*args, data=data, **kwargs)]
   1633         for line in lines:
   1634             self.add_line(line)

/envs/default/lib/python3.7/site-packages/matplotlib/axes/_base.py in __call__(self, data, *args, **kwargs)
    310                 this += args[0],
    311                 args = args[1:]
--> 312             yield from self._plot_args(this, kwargs)
    313 
    314     def get_next_color(self):

/envs/default/lib/python3.7/site-packages/matplotlib/axes/_base.py in _plot_args(self, tup, kwargs, return_kwargs)
    496 
    497         if x.shape[0] != y.shape[0]:
--> 498             raise ValueError(f"x and y must have same first dimension, but "
    499                              f"have shapes {x.shape} and {y.shape}")
    500         if x.ndim > 2 or y.ndim > 2:

ValueError: x and y must have same first dimension, but have shapes (100,) and (1,)

**Reply by AinhoaOleaga** [reply]
The code was : 
pye.plot_parameter_stability(ts = dataset)

**Reply by georgebv** [reply]
It's OK if you cannot provide original dataset. Perhaps you could create some dummy data which you can share with me to reproduce the error - it can be randomly generated as long as we use the same thing. Your error log doesn't help because this function works fine on my end for my test data: https://github.com/georgebv/pyextremes-notebooks/tree/master/data

I also advise using latest version - 2.2.4. Several important fixes were made since 2.0.0.

**Reply by AinhoaOleaga** [reply]
After downloading your repo to use the latest version everything is solved ! Thank you for your help :)
